File: install_git.js
const fs = require('fs');
const https = require('https');
const { execSync } = require('child_process');

const url = "https://github.com/git-for-windows/git/releases/download/v2.44.0.windows.1/MinGit-2.44.0-64-bit.zip";
const dest = "mingit.zip";

console.log("Downloading MinGit via Node...");
const file = fs.createWriteStream(dest);

https.get(url, (response) => {
    if (response.statusCode === 302) {
        https.get(response.headers.location, (res2) => {
            res2.pipe(file);
            file.on('finish', () => {
                file.close(() => {
                    console.log("Download complete. Extracting via powershell...");
                    try {
                        execSync('powershell.exe -Command "Expand-Archive -Path mingit.zip -DestinationPath portable_git -Force"', {stdio: 'inherit'});
                        console.log("Extracted to /portable_git. Ready.");
                    } catch (e) {
                        console.error("Shell extraction failed:", e.message);
                    }
                });
            });
        }).on('error', (err) => { fs.unlink(dest); console.error(err); });
    } else {
        response.pipe(file);
        // ...
    }
}).on('error', (err) => { 
    fs.unlink(dest); 
    console.error(err); 
});

File: install_git.py
import urllib.request
import zipfile
import os
import sys

URL = "https://github.com/git-for-windows/git/releases/download/v2.44.0.windows.1/MinGit-2.44.0-64-bit.zip"
ZIP_PATH = "mingit.zip"
EXTRACT_PATH = "portable_git"

print(f"Downloading {URL}...")
try:
    urllib.request.urlretrieve(URL, ZIP_PATH)
    print("Download complete.")
except Exception as e:
    print(f"Error downloading: {e}")
    sys.exit(1)

print(f"Extracting to {EXTRACT_PATH}...")
try:
    os.makedirs(EXTRACT_PATH, exist_ok=True)
    with zipfile.ZipFile(ZIP_PATH, 'r') as zip_ref:
        zip_ref.extractall(EXTRACT_PATH)
    print("Extraction complete.")
except Exception as e:
    print(f"Error extracting: {e}")
    sys.exit(1)
    
print("Cleaning up...")
try:
    os.remove(ZIP_PATH)
    print("Done. Portable Git is ready!")
except Exception as e:
    print(f"Cleanup error (ignored): {e}")
